packages/postcss-ordered-values/src/rules/grid.js

```diff
@@ -0,0 +1,83 @@
+export const normalizeGridAutoFlow = (gridAutoFlow) => {
+  let newValue = { front: '', back: '' };
+  let shouldNormalize = false;
+  gridAutoFlow.walk((node) => {
+    if (node.value === 'dense') {
+      shouldNormalize = true;
+      newValue.back = node.value;
+    } else if (['row', 'column'].includes(node.value.trim().toLowerCase())) {
+      shouldNormalize = true;
+      newValue.front = node.value;
+    } else {
+      shouldNormalize = false;
+    }
+  });
+  if (shouldNormalize) {
+    return `${newValue.front.trim()} ${newValue.back.trim()}`;
+  }
+  return gridAutoFlow;
+};
+
+export const normalizeGridColumnRowGap = (gridGap) => {
+  let newValue = { front: '', back: '' };
+  let shouldNormalize = false;
+  gridGap.walk((node) => {
+    // console.log(node);
+    if (node.value === 'normal') {
+      shouldNormalize = true;
+      newValue.front = node.value;
+    } else {
+      newValue.back = `${newValue.back} ${node.value}`;
+    }
+  });
+  if (shouldNormalize) {
+    return `${newValue.front.trim()} ${newValue.back.trim()}`;
+  }
+  return gridGap;
+};
+
+export const normalizeGridColumnRow = (grid) => {
+  // cant do normalization here using node, so copy it as a string
+  let gridValue = grid.toString(); // node -> string value
+  gridValue = gridValue.split('/'); // " 2 / 3 span " ->  [' 2','3 span ']
+  if (gridValue.length > 1) {
+    gridValue = gridValue.map((gridLine) => {
+      let normalizeValue = {
+        front: '',
+        back: '',
+      };
+      gridLine = gridLine.trimStart().trimEnd(); // '3 span ' -> '3 span'
+      gridLine.split(' ').forEach((node) => {
+        // ['3','span']
+        if (node === 'span') {
+          normalizeValue.front = node; // span _
+        } else {
+          normalizeValue.back = `${normalizeValue.back} ${node}`; // _ 3
+        }
+      });
+      return `${normalizeValue.front.trim()} ${normalizeValue.back.trim()}`; // span 3
+    });
+    // returns "2 / span 3"
+    return gridValue
+      .join(' / ')
+      .trimStart()
+      .trimEnd();
+  }
+  // doing this separating if `/` is not present as while joining('/') , it will add `/` at the end
+  gridValue = gridValue.map((gridLine) => {
+    let normalizeValue = {
+      front: '',
+      back: '',
+    };
+    gridLine = gridLine.trimStart().trimEnd();
```

Comment:
This is what `[String.prototype.trim](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim)` is for 🙂